lightning/src/routing/router.rs

```diff
@@ -222,46 +273,83 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		}
 	}
 
+	// For every new iteration of path finding, we will avoid using those channels,
+	// which had the lowest available amount in the previous iterations.
+	// It would help to have a strong diverse set of paths.
+	let mut channels_to_avoid = HashSet::new();
```

Comment:
Why do we need both this and used_channels_with_amounts_msat? We should be able to just ignore channels that have been fully utilized, no?